src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int GROW_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private float threshold;
+    private Node<K, V>[] hashTable;
+
+    public MyHashMap() {
+        hashTable = new Node[DEFAULT_CAPACITY];
+        threshold = DEFAULT_CAPACITY * LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size == threshold) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = hash(key);
+        if (hashTable[index] == null) {
+            hashTable[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> current = hashTable[index];
+        while (current != null) {
+            if (Objects.equals(current.key, key)) {
+                current.value = value;
+                return;
+            }
+            if (current.next == null) {
+                current.next = newNode;
+                size++;
+            }
+            current = current.next;
+        }
 
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> newNode = hashTable[hash(key)];
+        while (newNode != null) {
+            if (Objects.equals(newNode.key, key)) {
+                return newNode.value;
+            }
+            newNode = newNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        size = 0;
+        Node<K, V>[] oldTable = hashTable;
+        hashTable = new Node[DEFAULT_CAPACITY * GROW_FACTOR];
```

Comment:
The same problem like with threshold. Your new hashTable always will have capacity 32 (I mean, you will never receive capacity 64, 128, itc) because you calculate it throw multiplying of two constants, which never change.